utils/create-gnuplot-values-file.py

```diff
@@ -1,53 +1,58 @@
 import sys
 import lzma
-import statistics
 from os import listdir
-from os.path import isfile, join, basename
+from os.path import isfile, join, basename, splitext
+
+import pandas as pd
 
 ROUND_PRECISION = 2
 
-# Process an xz file
+columns = ["start_request",
+           "delay",
+           "status",
+           "written",
+           "read",
+           "method_and_url",
+           "thread_id",
+           "conn_id",
+           "conns",
+           "reqs",
+           "start",
+           "socket_writable",
+           "conn_est",
+           "tls_reuse",
+           "err"]
+
+
 def process(xz_file):
     with lzma.open(xz_file) as f:
-        lines = f.read().decode().strip('\n').split('\n')
-
-        first_request = lines[0].split(",")
-        last_request = lines[-1].split(",")
-        latencies = []
-        end_times = []
-        status = {}
-        errors = 0
-        for line in lines:
-            fields = line.split(",")
-            if len(fields) != 15:
-                raise Exception("Unknown line format [%s])" % line)
-
-            latencies.append(int(fields[1]))
-            end_times.append(int(fields[0]) + int(fields[1]))
-            if fields[2] in status:
-                status[fields[2]] += 1
-            else:
-                status[fields[2]] = 1
-
-            if fields[14]:
-                errors += 1
-
-        total_requests = len(latencies)
-        attack_duration = int(last_request[0]) - int(first_request[0])
-        request_rate = round(total_requests / (attack_duration/1e6),
-                             ROUND_PRECISION)
-        median = statistics.median(latencies) / 1e3
-        for code, i in status.items():
-            if code[0] == "4" or code[0] == 5:
-                errors += i
-
-        error_rate = round(errors / (attack_duration/1e6), ROUND_PRECISION)
-
-        print("%s %s %s %s" % (basename(xz_file)[0:-3], request_rate, median,
-                               error_rate))
-
-# We may want to reorder this afterwards
-directory = sys.argv[1]
-xz_files = sorted([f for f in listdir(directory) if f[-3:] == ".xz"])
-for f in xz_files:
-    process(join(directory, f))
+        df = pd.read_csv(f, names=columns, keep_default_na=False)
+
+    median = df['delay'].median() / 1e3
+
+    http_errors = df['status'].apply(lambda x: int(x / 100) in [4, 5])
+    total_http_errors = len(http_errors[http_errors==True])
+
+    conn_errors = (df['err'] != '')
+    total_conn_errors = len(conn_errors[conn_errors==True])
+
+    total_requests = len(df)
+    total_errors = total_http_errors + total_conn_errors
+
+    start_time = df['start_request'][total_requests - 1]
+    end_time = df['start_request'][0]
+    attack_duration = start_time - end_time
```

Comment:
`start_time` and `end_time` are reversed

Reply:
good catch! changing.

About setup.py, is it ok if I just add a requirements.txt for the moment?

Reply:
Yes, `requirements.txt` is ok